plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ApplyStaticNatVspCommand.java

```diff
@@ -51,13 +61,91 @@ public boolean isL3Network() {
         return _isL3Network;
     }
 
+    public boolean isVpc() {
+        return _isVpc;
+    }
+
     public List<Map<String, Object>> getStaticNatDetails() {
         return _staticNatDetails;
     }
 
+    public static class Builder implements CmdBuilder<ApplyStaticNatVspCommand> {
+        private String _networkDomainUuid;
+        private String _networkUuid;
+        private String _vpcOrSubnetUuid;
+        private boolean _isL3Network;
+        private boolean _isVpc;
+        private List<Map<String, Object>> _staticNatDetails;
+
+        public Builder networkDomainUuid(String networkDomainUuid) {
+            this._networkDomainUuid = networkDomainUuid;
+            return this;
+        }
+
+        public Builder networkUuid(String networkUuid) {
+            this._networkUuid = networkUuid;
+            return this;
+        }
+
+        public Builder vpcOrSubnetUuid(String vpcOrSubnetUuid) {
+            this._vpcOrSubnetUuid = vpcOrSubnetUuid;
+            return this;
+        }
+
+        public Builder isL3Network(boolean isL3Network) {
+            this._isL3Network = isL3Network;
+            return this;
+        }
+
+        public Builder isVpc(boolean isVpc) {
+            this._isVpc = isVpc;
+            return this;
+        }
+
+        public Builder staticNatDetails(List<Map<String, Object>> staticNatDetails) {
+            this._staticNatDetails = staticNatDetails;
+            return this;
+        }
+
+        @Override
+        public ApplyStaticNatVspCommand build() {
+            return new ApplyStaticNatVspCommand(_networkDomainUuid, _networkUuid, _vpcOrSubnetUuid, _isL3Network, _isVpc, _staticNatDetails);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Implemented <code>equals</code> method for the <code>Command</code> super class, calling it from subclass.